authors/apps/authentication/views.py

```diff
@@ -24,9 +32,49 @@ def post(self, request):
         # your own work later on. Get familiar with it.
         serializer = self.serializer_class(data=user)
         serializer.is_valid(raise_exception=True)
-        serializer.save()
-
-        return Response(serializer.data, status=status.HTTP_201_CREATED)
+        user = serializer.save()
+        token = serializer.instance.token
+
+        # Send a verification email to the user
+        # The combination of the token and uidb64 makes sure that the user
+        # has a unique verification link that expires in 7 days by default
+        subject = "Email Verification"
+        kwargs = {
+            'uidb64' : urlsafe_base64_encode(force_bytes(user.pk)).decode(),
+            'token' : email_activation_token.make_token(user)
+        }
+        verification_url = reverse(
+            'authentication:verify', kwargs=kwargs)
+        url = f"{request.scheme}://{request.get_host()}{verification_url}"
+        context = {'username': user.username,
+                   'url': url}
+        message = f"Dear {context['username']} \n Your account has successfully been created.\
+                    Please click the link below to activate your account. \n{context['url']}"
+        recipients = [user.email, ]
+        send_mail(subject, message,
+                  'ah.centauri@gmail.com', recipients)
+
+        return Response({
+            "token": token,
+            "message": f"A verification email has been sent to {user.email}",
+            "data": serializer.data
+        }, status=status.HTTP_201_CREATED)
+
+
+class VerifyEmailView(APIView):
+    def get(self, request, uidb64, token):
+        try:
+            uid = force_text(urlsafe_base64_decode(uidb64))
+            user = User.objects.get(pk=uid)
+        except User.DoesNotExist:
+            user = None
+        
+        if user is not None and email_activation_token.check_token(user, token):
+            user.email_confirmed = True
+            user.save()
+            return Response("Email successfully verified")
```

Comment:
I would _suggest_ adding key for the message - "Email successfully verified" and any other response messages you send.

Reply:
Let me work on it.